Katello/katello
# Fixes #34336 - CRUD simplified ACSs via the API

#### What are the changes introduced in this pull request?
Adds support for Simplified Alternate Content Sources. Users simply need to select which products they want to be included in an ACS and Katello will automatically maintain ACSs for every related repository (of the chosen content type) on all the selected smart proxies.  If repositories are created or destroyed, new ACSs will be created.

#### Considerations taken when implementing this change?
I'm updating hammer alongside to include the Products in the show command and also to reduce the amount of smart proxy parameters shown (https://github.com/Katello/hammer-cli-katello/pull/853).

I'm pretty set on keeping the content type requirement for simplified ACSs. Alternatively, we could just create ACSs for all supported types, but that makes things a bit complicated. Plus I think users will like having control over ignoring file repos, for example.

#### What are the testing steps for this pull request?
1) Create a couple smart proxies and a couple products
2) Create varying simplified smart proxies with different combinations of products and smart proxies
3) Try updating the simplified smart proxies in various ways, removing/adding smart proxies and products, etc
4) Delete simplified ACSs
5) Ensure that hammer works by testing this: https://github.com/Katello/hammer-cli-katello/pull/853
6) Refresh simplified ACSs and ensure that all the child ACSs were refreshed by checking in Dynflow and/or Pulp
7) Ensure that creating custom ACSs still works.

TODO:
- [x] Look into simplifying `app/lib/actions/katello/alternate_content_source/update.rb`
- [x] Audit simplified ACS products
- [x] Validations for fields like product and base url to ensure they don't exist for the wrong type
- [x] Check that HTTP proxy info from repositories isn't copied to ACS remotes.
- [x] Remove `product-names` param since there's no organizational scoping.
- [x] Don't create ACSs for repositories that don't have upstream URLs.
